ansible_events/drools/jpy_durable_rules_engine.py

```diff
@@ -0,0 +1,97 @@
+import json
+import logging
+
+import jpy
+import jpyutil
+
+_logger = logging.getLogger(__name__)
+
+
+class JpyDurableRulesEngine:
+    def __init__(self, classpath):
+        if not isinstance(classpath, list):
+            classpath = [classpath]
+
+        jpyutil.init_jvm(jvm_maxmem="512M", jvm_classpath=classpath)
+
+        self._executor_class = jpy.get_type(
+            "org.drools.yaml.core.RulesExecutor"
+        )
+        self._durable_rule_match_class = jpy.get_type(
+            "org.drools.yaml.durable.domain.DurableRuleMatch"
+        )
+        self._durable_notation = jpy.get_type(
+            "org.drools.yaml.durable.DurableNotation"
+        ).INSTANCE
+
+        self._executor_container = jpy.get_type(
+            "org.drools.yaml.core.RulesExecutorContainer"
+        ).INSTANCE
+        self._last_response = None
+
+    def create_ruleset(self, ruleset_name, ruleset_string):
+        payload = {ruleset_name: json.loads(ruleset_string)}
+        s = json.dumps(payload)
+        executor = self._executor_class.createFromJson(
+            self._durable_notation, s
+        )
+        return executor.getId()
+
+    def assert_event(self, session_id, serialized_fact):
+        def command(fact):
+            executor = self._executor_container.get(session_id)
+            return executor.processEvents(fact)
+
+        return self._process_message(session_id, serialized_fact, command)
+
+    def assert_fact(self, session_id, serialized_fact):
+        def command(fact):
+            executor = self._executor_container.get(session_id)
+            return executor.processFacts(fact)
+
+        return self._process_message(session_id, serialized_fact, command)
+
+    def _process_message(self, session_id, serialized_fact, command):
+        fact = json.loads(serialized_fact)
+        if "j" in serialized_fact:
```

Comment:
@evacchi Is this a leftover code setting the j fact?

Reply:
I think this was "fixing" some issue on the Java end and must have ended up in the final code.